### PR TITLE
[signer] Add block_proposal_max_age_secs signer configuration to drop old proposals without further processing

### PR DESCRIPTION
**Pull request by jferrant**
Closes https://github.com/stacks-network/stacks-core/issues/5513

### CODE REVIEW

File: stacks-signer/src/v0/signer.rs [hunk 1]
```diff
@@ -344,6 +347,23 @@ impl Signer {
...
+        if block_proposal
+            .block
+            .header
+            .timestamp
+            .saturating_add(self.block_proposal_max_age_secs)
+            < get_epoch_time_secs()
+        {
+            // Block is too old. Drop it with a warning. Don't even bother broadcasting to the node.
+            warn!("{self}: Received a block proposal that is more than {} secs old. Ignoring...", self.block_proposal_max_age_secs;
+                "block_id" => %block_proposal.block.block_id(),
```
Comment (obycode) [inline stacks-signer/src/v0/signer.rs:359]: Could you include the `signer_sighash` in here? I often am looking through the signer logs for that hash.

**Review by obycode — DISMISSED** [review]
LGTM! And thanks for the general improvement to the existing integration tests!!

File: stacks-signer/src/v0/signer.rs (hunk 1 above)
Comment (jferrant) [reply stacks-signer/src/v0/signer.rs:359]: sure thing!

File: stacks-signer/src/v0/signer.rs [hunk 2]
```diff
@@ -356,6 +356,7 @@ impl Signer {
         {
             // Block is too old. Drop it with a warning. Don't even bother broadcasting to the node.
             warn!("{self}: Received a block proposal that is more than {} secs old. Ignoring...", self.block_proposal_max_age_secs;
+                "signer_signature_hash" => %block_proposal.block.header.signer_signature_hash(),
```
Comment (obycode) [inline stacks-signer/src/v0/signer.rs:359]: Nit: all of the other structured logs in this file call it `signer_sighash`. It's probably good for debugging if they all use the same name.

Comment (jferrant) [reply stacks-signer/src/v0/signer.rs]: Done!

**Review by obycode — APPROVED** [review]

File: stacks-signer/src/config.rs [hunk 3]
```diff
@@ -39,6 +39,7 @@ const BLOCK_PROPOSAL_VALIDATION_TIMEOUT_MS: u64 = 120_000;
 const DEFAULT_FIRST_PROPOSAL_BURN_BLOCK_TIMING_SECS: u64 = 60;
 const DEFAULT_TENURE_LAST_BLOCK_PROPOSAL_TIMEOUT_SECS: u64 = 30;
 const TENURE_IDLE_TIMEOUT_SECS: u64 = 300;
+const DEFAULT_BLOCK_PROPOSAL_MAX_AGE_SECS: u64 = 600;
```
Comment (jcnelson) [inline stacks-signer/src/config.rs:42]: This is a long time. Are there consequences to making it shorter -- i.e. a small integer multiple of the expected RTT for the peer network?

**Review by jcnelson — APPROVED** [review]

File: stacks-signer/src/config.rs (hunk 3 above)
Comment (jferrant) [reply stacks-signer/src/config.rs:42]: I am totally fine with making it shorter. This was just pulled out of my butt ...

I am not sure what we would consider a good base...or even what the expected RTT is

Comment (jferrant) [reply stacks-signer/src/config.rs:42]: Made a ticket to look into a better default value. https://github.com/stacks-network/stacks-core/issues/5566

**Pull request merged**